distributed/preloading.py

```diff
@@ -0,0 +1,87 @@
+import atexit
+import shutil
+import logging
+import os
+import sys
+from importlib import import_module
+
+from .compatibility import reload, invalidate_caches, cache_from_source
+
+
+logger = logging.getLogger(__name__)
+
+
+def load_file(path):
+    """ Loads modules for a file (.py, .pyc, .zip, .egg) """
+    directory, filename = os.path.split(path)
+    name, ext = os.path.splitext(filename)
+    names_to_import = []
+    if ext in ('.py', '.pyc'):
+        if directory not in sys.path:
+            sys.path.insert(0, directory)
```

Comment:
For all current uses the directory already is on the path. The only reason I added this is that this function takes care of the path for `zip` and `egg`. I will just remove this line an everything will still work, only the behaviour will be less consistent: "this function takes care of the path if you pass `zip` or `egg`, but for `py` and `pyc` you need to do it yourself.